mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -91,8 +91,8 @@ private[classification] trait LogisticRegressionParams extends ProbabilisticClas
   @Since("2.1.0")
   final val family: Param[String] = new Param(this, "family",
     "The name of family which is a description of the label distribution to be used in the " +
-      s"model. Supported options: ${supportedFamilyNames.mkString(", ")}.",
-    ParamValidators.inArray[String](supportedFamilyNames))
+      s"model (case-insensitive). Supported options: ${supportedFamilyNames.mkString(", ")}.",
```

Comment:
Searching through the code base these are the places where we use Param[String]:

spark-mllib_2.11
org.apache.spark.ml.classification
LogisticRegression.scala
  final val family: Param[String] = new Param(this, "family",
MultilayerPerceptronClassifier.scala
  final val solver: Param[String] = new Param[String](this, "solver",
  final val solver: Param[String] = new Param[String](this, "solver",
NaiveBayes.scala
  final val modelType: Param[String] = new Param[String](this, "modelType", "The model type " +
  final val modelType: Param[String] = new Param[String](this, "modelType", "The model type " +
org.apache.spark.ml.clustering
KMeans.scala
  final val initMode = new Param[String](this, "initMode", "The initialization algorithm. " +
LDA.scala
  final val optimizer = new Param[String](this, "optimizer", "Optimizer or inference" +
  final val topicDistributionCol = new Param[String](this, "topicDistributionCol", "Output column" +
org.apache.spark.ml.evaluation
BinaryClassificationEvaluator.scala
  val metricName: Param[String] = {
MulticlassClassificationEvaluator.scala
  val metricName: Param[String] = {
RegressionEvaluator.scala
  val metricName: Param[String] = {
org.apache.spark.ml.feature
Bucketizer.scala
  val handleInvalid: Param[String] = new Param[String](this, "handleInvalid", "how to handle " +
  val handleInvalid: Param[String] = new Param[String](this, "handleInvalid", "how to handle " +
ChiSqSelector.scala
  final val selectorType = new Param[String](this, "selectorType",
QuantileDiscretizer.scala
  val handleInvalid: Param[String] = new Param[String](this, "handleInvalid", "how to handle " +
  val handleInvalid: Param[String] = new Param[String](this, "handleInvalid", "how to handle " +
RFormula.scala
  val formula: Param[String] = new Param(this, "formula", "R model formula")
SQLTransformer.scala
  final val statement: Param[String] = new Param[String](this, "statement", "SQL statement")
  final val statement: Param[String] = new Param[String](this, "statement", "SQL statement")
Tokenizer.scala
  val pattern: Param[String] = new Param(this, "pattern", "regex pattern used for tokenizing")
org.apache.spark.ml.param
ParamsSuite.scala
      val param = new Param[String](dummy, "name", "doc")
org.apache.spark.ml.param.shared
sharedParams.scala
  final val featuresCol: Param[String] = new Param[String](this, "featuresCol", "features column name")
  final val featuresCol: Param[String] = new Param[String](this, "featuresCol", "features column name")
  final val labelCol: Param[String] = new Param[String](this, "labelCol", "label column name")
  final val labelCol: Param[String] = new Param[String](this, "labelCol", "label column name")
  final val predictionCol: Param[String] = new Param[String](this, "predictionCol", "prediction column name")
  final val predictionCol: Param[String] = new Param[String](this, "predictionCol", "prediction column name")
  final val rawPredictionCol: Param[String] = new Param[String](this, "rawPredictionCol", "raw prediction (a.k.a. confidence) column name")
  final val rawPredictionCol: Param[String] = new Param[String](this, "rawPredictionCol", "raw prediction (a.k.a. confidence) column name")
... P...
... P...
  final val varianceCol: Param[String] = new Param[String](this, "varianceCol", "Column name for the biased sample variance of prediction")
  final val varianceCol: Param[String] = new Param[String](this, "varianceCol", "Column name for the biased sample variance of prediction")
  final val inputCol: Param[String] = new Param[String](this, "inputCol", "input column name")
  final val inputCol: Param[String] = new Param[String](this, "inputCol", "input column name")
  final val outputCol: Param[String] = new Param[String](this, "outputCol", "output column name")
  final val outputCol: Param[String] = new Param[String](this, "outputCol", "output column name")
... P...
... P...
  final val weightCol: Param[String] = new Param[String](this, "weightCol", "weight column name. If this is not set or empty, we treat all instance weights as 1.0")
  final val weightCol: Param[String] = new Param[String](this, "weightCol", "weight column name. If this is not set or empty, we treat all instance weights as 1.0")
  final val solver: Param[String] = new Param[String](this, "solver", "the solver algorithm for optimization. If this is not set or empty, default value is 'auto'")
  final val solver: Param[String] = new Param[String](this, "solver", "the solver algorithm for optimization. If this is not set or empty, default value is 'auto'")
org.apache.spark.ml.recommendation
ALS.scala
  val userCol = new Param[String](this, "userCol", "column name for user ids. Ids must be within " +
  val itemCol = new Param[String](this, "itemCol", "column name for item ids. Ids must be within " +
  val ratingCol = new Param[String](this, "ratingCol", "column name for ratings")
  val intermediateStorageLevel = new Param[String](this, "intermediateStorageLevel",
  val finalStorageLevel = new Param[String](this, "finalStorageLevel",
org.apache.spark.ml.regression
AFTSurvivalRegression.scala
  final val censorCol: Param[String] = new Param(this, "censorCol", "censor column name")
  final val quantilesCol: Param[String] = new Param(this, "quantilesCol", "quantiles column name")
GeneralizedLinearRegression.scala
  final val family: Param[String] = new Param(this, "family",
  final val link: Param[String] = new Param(this, "link", "The name of link function " +
  final val linkPredictionCol: Param[String] = new Param[String](this, "linkPredictionCol",
  final val linkPredictionCol: Param[String] = new Param[String](this, "linkPredictionCol",
org.apache.spark.ml.tree
treeParams.scala
  final val impurity: Param[String] = new Param[String](this, "impurity", "Criterion used for" +
  final val impurity: Param[String] = new Param[String](this, "impurity", "Criterion used for" +
  final val impurity: Param[String] = new Param[String](this, "impurity", "Criterion used for" +
  final val impurity: Param[String] = new Param[String](this, "impurity", "Criterion used for" +
  final val featureSubsetStrategy: Param[String] = new Param[String](this, "featureSubsetStrategy",
  final val featureSubsetStrategy: Param[String] = new Param[String](this, "featureSubsetStrategy",
  val lossType: Param[String] = new Param[String](this, "lossType", "Loss function which GBT" +
  val lossType: Param[String] = new Param[String](this, "lossType", "Loss function which GBT" +
  val lossType: Param[String] = new Param[String](this, "lossType", "Loss function which GBT" +
  val lossType: Param[String] = new Param[String](this, "lossType", "Loss function which GBT" +
org.apache.spark.ml.util
DefaultReadWriteTest.scala
  final val stringParam: Param[String] = new Param[String](this, "stringParam", "doc")
  final val stringParam: Param[String] = new Param[String](this, "stringParam", "doc")